advanced/am-pattern-2/templates/wso2am-pattern-2-secrets.yaml

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+  {{ if and (not .Values.wso2.subscription.username) (not .Values.wso2.subscription.password) }}
+
+    {{- $username := .Values.wso2.subscription.username }}
+    {{- $password := .Values.wso2.subscription.password }}
+    {{- $email := .Values.wso2.subscription.username }}
+    {{- $regId := default "docker.wso2.com" .Values.wso2.dockerRegistry }}
+    {{- $auth := printf "%s:%s" $username $password | b64enc }}
+    {{- $files := .Files }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: wso2am-pattern-2-wso2-private-registry-creds
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ $files.Get "auth.json" | replace "reg.id" $regId | replace "reg.username" $username | replace "reg.password" $password | replace "reg.email" $email | replace "reg.auth" $auth | b64enc }}
+
+  {{ end }}
```

Comment:
@Chathuru please add an EOF line.

Reply:
Fix the issue in https://github.com/wso2/kubernetes-apim/pull/344/commits/c1d371588b9d9e50fa267a4bbf5a74bc7b50517b